ipfs-9-to-10/migration/config_conv.go

```diff
@@ -0,0 +1,189 @@
+package mg9
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	log "github.com/ipfs/fs-repo-migrations/stump"
+)
+
+var (
+	ip4BootstrapAddr  = "/ip4/104.131.131.82/tcp/4001/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+	quicBootstrapAddr = "/ip4/104.131.131.82/udp/4001/quic/p2p/QmaCpDMGvV2BGHeYERUEnRQAwe3N8SzbUtfsmvsqQLuvuJ"
+)
+
+// convArray does an inplace conversion of an array of strings
+// configuration from one version to another
+type convArray func([]string) []string
+
+// convertFile converts a config file from one version to another
+func convertFile(orig string, new string, enableQuic bool, convBootstrap convArray, convSwarm convArray) error {
+	in, err := os.Open(orig)
+	if err != nil {
+		return err
+	}
+	out, err := os.Create(new)
```

Comment:
missing `defer xx.Close()` ?

Reply:
Also, doublecheck that the permissions are correct. If this is the configuration, it should probably be 0600 ? But now that I check, my ipfs config has 0644 :/ Did we mess it up in a previous migration?